### ISSUE TITLE
Definite assignment assertion in JS not flagged

### ISSUE THREAD

**Issue by Josh-Cena** [microsoft/TypeScript#48446]
# Bug Report

Noted by someone on Discord.

### 🔎 Search Terms

non null assertion javascript error

Related https://github.com/microsoft/TypeScript/issues/17222

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?filetype=js#code/MYewdgzgLgBAhjAvDA3vAhALhgRhgXwG4AoIA)

### 💻 Code

```ts
const a = { a!: 1 };
```

### 🙁 Actual behavior

There's no error; crashes at runtime

### 🙂 Expected behavior

There should be an error

**Comment by DanielRosenwasser** [comment]
Uh, forget JavaScript - does it ever make sense to have a definite assignment assertion on an object property declaration?

```ts
let obj = {
    wat!: 123
}
```

The **only** instance I can actually think of is on a shorthand property which would act more as a non-null assertion than a definite assignment assertion... but ironically, we issue an error if you try to do that!

```ts
declare function tryGetNumber(): number | undefined;

let num = tryGetNumber();
let obj = {
    num!
//     ~
// A definite assignment assertion '!' is not permitted in this context.
};
```

**Comment by Josh-Cena** [comment]
Yeah, someone on the TS Discord pointed this out. I never would imagine this being valid :/

**Comment by Josh-Cena** [comment]
So my original post is already an error (and seems it has been for a while) since TS 4.8. However, there's still not specialized check in JS files for the following:

```js
class A {
  a!;
}
```

Could this issue be about making the above an error?
